docs/developer/proposal/EP_TEMPLATE.md

```diff
@@ -0,0 +1,235 @@
+[Show the proposed changes to the main configuration file (e.g., `config.go` or `config.yaml`).]
+
+```go
+// Example config struct change
+type NewFeatureConfig struct {
+    Enabled    *bool  `yaml:"enabled"`
+    SomeValue  string `yaml:"someValue"`
+}
+```
+
+### New Configuration File (if applicable)
+
+[If a new configuration file is needed, describe its format and provide an example.]
+
+```yaml
+# Example: /etc/kepler/new-feature.yaml
+feature:
+  key: value
+```
+
+### Security Considerations
+
+[Detail the security implications of this enhancement. How are secrets managed? What are the authentication/authorization requirements?]
+
+## Deployment Examples
+
+[Provide examples of how to deploy and use this feature in different environments (e.g., Kubernetes, standalone).]
+
+### Kubernetes Environment
+
+[Show a sample Kubernetes manifest (e.g., DaemonSet, Deployment) with the new configuration.]
+
+### Standalone Deployment
+
+[Provide command-line examples for running the application with the new feature enabled.]
+
+## Testing Strategy
+
+### Test Coverage
+
+[Describe the testing plan. What types of tests will be added (unit, integration, end-to-end)? What is the target code coverage?]
+
+- **Unit Tests**: ...
+- **Integration Tests**: ...
+- **End-to-End Tests**: ...
+
+### Test Infrastructure
+
+[Describe any new infrastructure or tools required for testing (e.g., simulators, mock servers).]
+
+## Migration and Compatibility
+
+### Backward Compatibility
+
+[Explain how this change affects backward compatibility. Are there any breaking changes? How will they be managed?]
+
+### Migration Path
+
+[Provide a step-by-step guide for users to migrate from the old system to the new one.]
+
+1. **Phase 1**: ...
+2. **Phase 2**: ...
+3. ...
+
+### Rollback Strategy
+
+[Describe the process for rolling back the change if issues are discovered after deployment.]
+
+## Metrics Output
+
+[List any new or modified Prometheus metrics that will be exposed by this feature.]
+
+```prometheus
+# Description of new metric
+new_metric_name{label="value"} 123
+```
+
+## Implementation Plan
+
+[Break down the implementation into a series of phases or milestones.]
+
+### Phase 1: Foundation
+
+- Task 1...
+- Task 2...
+
+### Phase 2: Core Functionality
+
+- Task 1...
+- Task 2...
+
+### Phase 3: Testing and Documentation
+
+- Task 1...
+- Task 2...
+
+## Risks and Mitigations
+
+[Identify potential risks (technical, operational) and propose mitigation strategies for each.]
+
+### Technical Risks
+
+- **Risk**: ...
+  - **Mitigation**: ...
+
+### Operational Risks
+
+- **Risk**: ...
+  - **Mitigation**: ...
+
+## Alternatives Considered
+
+[Describe any alternative solutions that were considered and explain why they were rejected.]
+
+### Alternative 1: [Name of Alternative]
+
+- **Description**: ...
+- **Reason for Rejection**: ...
+
+### Alternative 2: [Name of Alternative]
+
+- **Description**: ...
+- **Reason for Rejection**: ...
+
+## Success Metrics
```

Comment:
do we need this field? maybe just if accpetable?

Reply:
yes, and no. We can use it where it makes sense. I will submit few proposals for reference.